src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,105 @@
+"""
+Passkey Tests.
+
+:requirement: passkey
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.tier(0)
+@pytest.mark.topology(KnownTopology.Client)
+def test_passkey__register__sssctl(client: Client, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a key with sssctl
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use sssctl to register a FIDO2 key.
+        2. Check the output.
+    :expectedresults:
+        1. New key is registered
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = client.sssctl.passkey_register(
+        username="user1",
+        domain="ldap.test",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script",
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip(), "Failed to register a key with sssctl"
+
+
+@pytest.mark.tier(0)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__register__ipa(ipa: IPA, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a passkey with the IPA command
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use ipa command to register a FIDO2.
+        2. Check the output that contains the user key mapping data.
+    :expectedresults:
+        1. New key is registered with IPA command.
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = (
+        ipa.user("user1")
+        .add()
+        .passkey_add_register(
+            pin=123456,
+            device=f"{moduledatadir}/umockdev.device",
+            ioctl=f"{moduledatadir}/umockdev.ioctl",
+            script=f"{testdatadir}/umockdev.script",
+        )
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip(), "Failed to register a key with the IPA command"
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.AD)
+def test_passkey__su(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication of user with LDAP and IPA
+    :setup:
+        1. Add a LDAP and IPA user with passkey_mapping.
+        2. Setup SSSD client with FIDO.
+        3. Create the recording files when authentications is successful using umockdev.
```

Comment:
yes thats why I added in setup part not in test case steps.

Reply:
Question is if it should provide information about how the automation works (use umockdev, create recordings, upload recordings, prepare script to run umockdev, ...) or how to rerun the test manually. IMHO we should focus on the latter, otherwise it would get overly complicated.

Reply:
Yes, but in my opinion, this is not enough information to successfully reproduce the test using umockdev, right? But I will leave it up to you (and Iker) or in general on how QA is using this information. I don't have strong opinion on this.

Reply:
From my point of view we should explain from a high level perspective what the test does, but I'll leave up to QE what they are expecting from these steps.

In any case, I don't see where step 3 of setup is happening. Once sssd starts (setup 2), the test checks the user authentication (steps 1).

Reply:
I am okay with both, @spoore1 what do you think?

Reply:
@madhuriupadhye   I think we can leave that step out and modify step 2 in setup to add `and umockdev setup` to step 2 so it better matches earlier tests' docstrings.    I agree with @pbrezina that it's not enough info to fully configure a system to reproduce the scenario.  We can include that the test documentation.   I think the `umockdev setup` note earlier is enough IMO.